lib/src/signed_video_h26x_auth.c

```diff
@@ -964,3 +968,36 @@ signed_video_add_nalu_and_authenticate(signed_video_t *self,
 
   return svi_rc_to_signed_video_rc(status);
 }
+
+SignedVideoReturnCode
+signed_video_set_public_key(signed_video_t *self, const char *public_key, size_t public_key_size)
+{
+  if (!self || !public_key || public_key_size == 0) return SV_INVALID_PARAMETER;
+  if (self->signature_info->public_key) return SV_NOT_SUPPORTED;
+
+  // Return SV_NOT_SUPPORTED if not called from start of stream
+  if (self->authentication_started) return SV_NOT_SUPPORTED;
+
+
+  sign_algo_t *algo = &(self->signature_info->algo);
+  svi_rc status = SVI_UNKNOWN;
+  SVI_TRY()
+    SVI_THROW(openssl_get_algo_of_public_key(public_key, public_key_size, algo));
+    // Allocate memory and copy |public_key|.
+    self->signature_info->public_key = malloc(public_key_size);
+    SVI_THROW_IF(!self->signature_info->public_key, SVI_MEMORY);
+    memcpy(self->signature_info->public_key, public_key, public_key_size);
+
+    self->signature_info->public_key_size = public_key_size;
+    self->has_public_key = true;
+
+  SVI_CATCH()
+  {
+    // Remove key upon failure.
+    free(self->signature_info->public_key);
+    self->signature_info->public_key = NULL;
+  }
```

Comment:
It is now possible to remove this whole CATCH part. If we fail somewhere above, public_key is already NULL.